tests/test_surface/test_new_surface.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+import numpy as np
+import hypothesis.strategies as st
+from hypothesis import given
+from xtgeo.surface.regular_surface import RegularSurface
+
+
+def test_default_values():
+    surf = RegularSurface(2, 2, 0.0, 0.0)
+    assert surf.values.data.tolist() == [[0.0, 0.0], [0.0, 0.0]]
+
+
+@pytest.mark.parametrize(
+    "input_val, expected_result",
+    [
+        (1, [[1.0, 1.0], [1.0, 1.0]]),
+        ([1, 2, 3, 4], [[1.0, 2.0], [3.0, 4.0]]),
+        (np.zeros((2, 2)), [[0.0, 0.0], [0.0, 0.0]]),
+        (None, [[0.0, 0.0], [0.0, 0.0]]),
+        (
+            np.ma.MaskedArray([[2, 2], [2, 2]], mask=[[True, False], [False, True]]),
+            [[2.0, 2.0], [2.0, 2.0]],
+        ),
+    ],
+)
+def test_values_type(input_val, expected_result):
+    surf = RegularSurface(2, 2, 0.0, 0.0, values=input_val)
+    assert isinstance(surf.values, np.ma.MaskedArray)
+    assert surf.values.data.tolist() == expected_result
+
+
+@pytest.mark.parametrize("input_values", [None, 2])
+@pytest.mark.parametrize(
+    "input_val, expected_result",
+    [
+        (1, [[1.0, 1.0], [1.0, 1.0]]),
+        ([1, 2, 3, 4], [[1.0, 2.0], [3.0, 4.0]]),
+        (np.zeros((2, 2)), [[0.0, 0.0], [0.0, 0.0]]),
+        (np.ma.zeros((2, 2)), [[0.0, 0.0], [0.0, 0.0]]),
+        (np.ma.ones((2, 2)), [[1.0, 1.0], [1.0, 1.0]]),
+    ],
+)
+def test_values_setter(input_values, input_val, expected_result):
+    surf = RegularSurface(2, 2, 0.0, 0.0, values=input_values)
+    surf.values = input_val
+    assert surf.values.data.tolist() == expected_result
+
+
+@pytest.mark.parametrize(
+    "input_val, expected_data, expected_mask",
+    [
+        (1, [2.0, 1.0, 1.0, 2.0], [[True, False], [False, True]]),
+        ([1, 2, 3, 4], [1.0, 2.0, 3.0, 4.0], [[False, False], [False, False]]),
+        ([1, 1e33, 3, 4], [1.0, 1e33, 3.0, 4.0], [[False, True], [False, False]]),
+        (np.zeros((2, 2)), [0.0, 0.0, 0.0, 0.0], [[False, False], [False, False]]),
+        (np.ma.zeros((2, 2)), [0.0, 0.0, 0.0, 0.0], [[False, False], [False, False]]),
+        (np.ma.ones((2, 2)), [1.0, 1.0, 1.0, 1.0], [[False, False], [False, False]]),
+    ],
+)
+def test_values_mask_setter(input_val, expected_data, expected_mask):
+    surf = RegularSurface(
+        2,
+        2,
+        0.0,
+        0.0,
+        values=np.ma.MaskedArray([[2, 2], [2, 2]], mask=[[True, False], [False, True]]),
+    )
+    surf.values = input_val
+    assert surf.values.mask.tolist() == expected_mask
+    assert list(surf.values.data.flatten()) == expected_data
+
+
+@given(data=st.lists(st.floats(allow_nan=False), min_size=100, max_size=100))
+def test_input_lists(data):
+    surf = RegularSurface(10, 10, 0.0, 0.0, values=data)
+    assert surf.values.data.flatten().tolist() == pytest.approx(data)
+
+
+@given(data=st.lists(st.floats(allow_nan=False)))
+def test_wrong_size_input_lists(data):
+    if len(data) == 100:
```

Comment:
not to start an early review here, but you can use `assume len(data) == 100`, which would signify to hypothesis that the created dataset is not valid.